core/src/main/java/io/github/fourlastor/game/coordinates/Hex.java

```diff
@@ -65,4 +64,57 @@ public void set(Hex other) {
     public boolean sameAxisAs(Hex hex) {
         return hex.cube.x == cube.x || hex.cube.y == cube.y || hex.cube.z == cube.z;
     }
+
+    public Hex offset(Direction direction, int amount) {
+        return new Hex(cube.cpy().add(direction.cube.cpy().scl(amount)));
+    }
+
+    public enum Direction {
+        S(0, -1, 1),
+        SE(1, -1, 0),
+        NE(1, 0, -1),
+        N(0, +1, -1),
+        NW(-1, 1, 0),
+        SW(-1, 0, 1);
+
+        private final GridPoint3 cube;
+
+        Direction(int x, int y, int z) {
+            cube = new GridPoint3(x, y, z);
+        }
+
+        public static Direction byValue(GridPoint3 cube) {
+            return Arrays.stream(values())
+                    .filter(dir -> dir.cube.equals(cube))
+                    .findFirst()
+                    .get();
+        }
+
+        public static Direction byName(String value) {
+
+            Optional<Direction> test = Arrays.stream(values())
+                    .filter(dir -> dir.name().equalsIgnoreCase(value))
+                    .findFirst();
+            return test.orElse(null);
```

Comment:
I'd change this to `test.orElseThrow(() -> new IllegalArgumentException("No direction found for name " + value))`